src/parser.js

```diff
@@ -1477,12 +1470,9 @@ export default class Parser {
     let jsonSchema;
 
     try {
-      jsonSchema = convertSchema(schema);
-    } catch (exception) {
-      this.createAnnotation(
-        annotations.DATA_LOST, this.path,
-        'Circular references in schema are not yet supported',
-      );
+      jsonSchema = convertSchema(schema, this.swagger);
+    } catch (error) {
+      this.createAnnotation(annotations.VALIDATION_ERROR, this.path, error.message);
```

Comment:
The old code-path here was actually catching other errors that are not for circular referencing. For example before if there was external reference it would show the wrong error.